jupyterhub/auth.py

```diff
@@ -281,8 +281,8 @@ def check_blacklist(self, username):
                 'name': authenticated,
             }
         authenticated.setdefault('auth_state', None)
-        admin_status = await maybe_future(self.is_admin(handler, data))
-        authenticated.setdefault('admin', admin_status)
+        # Leave the default as None, but reevaluate later post-whitelist
+        authenticated.setdefault('admin', None)
```

Comment:
I figured leaving a default of None at the beginning was safest, instead of leaving it undefined until the end.

If other `Authenticator` functions start to use the `authenticated` dict, having a default set as soon as possible seems best.